carma-messenger-core/cpp_message/src/MobilityOperation_Message.cpp

```diff
@@ -0,0 +1,257 @@
+        uint8_t string_content_targetId[string_size];
+        for(auto i=0;i<string_size;i++)
+        {
+            string_content_targetId[i]=plainMessage.header.recipient_id[i];
+        }
+        message->value.choice.TestMessage03.header.targetStaticId.buf=string_content_targetId;
+        message->value.choice.TestMessage03.header.targetStaticId.size=string_size;
+        
+         //convert bsm_id string to char array
+        string_size=plainMessage.header.sender_bsm_id.size();
+        uint8_t string_content_BSMId[string_size];
+        for(auto i=0;i<string_size;i++)
+        {
+            string_content_BSMId[i]=plainMessage.header.sender_bsm_id[i];
+        }
+        message->value.choice.TestMessage03.header.hostBSMId.buf=string_content_BSMId;
+        message->value.choice.TestMessage03.header.hostBSMId.size=string_size;
+        
+         //convert plan_id string to char array
+        string_size=plainMessage.header.plan_id.size();
+        uint8_t string_content_planId[string_size];
+        for(auto i=0;i<string_size;i++)
+        {
+            string_content_planId[i]=plainMessage.header.plan_id[i];
+        }
+        message->value.choice.TestMessage03.header.planId.buf=string_content_planId;
+        message->value.choice.TestMessage03.header.planId.size=string_size;
+        //get timestamp and convert to char array
+        uint64_t time=plainMessage.header.timestamp;
+        std::string timestamp=std::to_string(time);
+        string_size=timestamp.size();
+        uint8_t string_content_timestamp[string_size];
+        for(auto i=0;i<string_size;i++)
+        {
+            string_content_timestamp[i]=timestamp[i];
+        }
+        message->value.choice.TestMessage03.header.timestamp.buf=string_content_timestamp;
+        message->value.choice.TestMessage03.header.timestamp.size=string_size;
+
+         //convert strategy string to char array
+        string_size=plainMessage.strategy.size();
+        uint8_t string_content_strategy[string_size];
+        for(auto i=0;i<string_size;i++)
+        {
+            string_content_strategy[i]=plainMessage.strategy[i];
+        }
+        message->value.choice.TestMessage03.body.strategy.buf=string_content_strategy;
+        message->value.choice.TestMessage03.body.strategy.size=string_size;
+        
+
+        //convert parameters string to char array
+        string_size=plainMessage.strategy_params.size();
+        uint8_t string_content_params[string_size];
+        for(auto i=0;i<string_size;i++)
+        {
+            string_content_params[i]=plainMessage.strategy_params[i];
+        }
+        message->value.choice.TestMessage03.body.operationParams.buf=string_content_params;
+        message->value.choice.TestMessage03.body.operationParams.size=string_size;
+        
+        //encode message
+        ec=uper_encode_to_buffer(&asn_DEF_MessageFrame, 0 , message , buffer , buffer_size);
+         
+        //log a warning if that fails
```

Comment:
Mention of logging a warning, but no actual log statement.

Reply:
Added warning